txpipe/map_tests.py

```diff
@@ -0,0 +1,162 @@
+
+        if not (kappa.size == ngal.size):
+            raise ValueError("Maps are different sizes")
+
+        output_dir = self.open_output("map_systematic_correlations", wrapper=True)
+
+
+        root = self.config["supreme_path_root"]
+        sys_maps = glob.glob(f"{root}*.hs")
+        nsys = len(sys_maps)
+        print(f"Found {nsys} total systematic maps")
+
+        outputs = []
+        for i, map_path in enumerate(sys_maps):
+            # strip root, .hs, and underscores to get friendly name
+            sys_name = map_path[len(root):-3].strip("_")
+
+            # get actual data for this map
+            sys_map = self.read_healsparse(map_path, nside)
+
+            # Correlate with g1, g2, ngal, kappa
+            print(f"Correlating systematic {i+1}/{nsys} {sys_name}")
+            corr = self.correlate(sys_map, source_g1, mask)
+            outfile = self.save(sys_name, 'g1', corr, output_dir)
+            outputs.append(outfile)
+            
+            corr = self.correlate(sys_map, source_g2, mask)
+            outfile = self.save(sys_name, 'g2', corr, output_dir)
+            outputs.append(outfile)
+            
+            corr = self.correlate(sys_map, ngal, mask)
+            outfile = self.save(sys_name, 'number_density', corr, output_dir)
+            outputs.append(outfile)
+            
+            corr = self.correlate(sys_map, kappa, mask)
+            outfile = self.save(sys_name, 'convergence', corr, output_dir)
+            outputs.append(outfile)
+
+
+        output_dir.write_listing(outputs)
+
+    def correlate(self, sys_map, data_map, mask):
+        import scipy.stats
+        import healpy
+
+        N = self.config["nbin"]
+        f = 0.5 * self.config["outlier_fraction"]
+
+        # clean the data
+        finite = (
+            np.isfinite(sys_map + data_map)
+            & (sys_map != healpy.UNSEEN)
+            & (data_map != healpy.UNSEEN)
+            & (mask > 0)
+        )
+
+        sys_map = sys_map[finite]
+        data_map = data_map[finite]
+
+        # Choose bin edges and put pixels in them.
+        percentiles = np.linspace(f, 1 - f, N + 1)
+        bin_edges = scipy.stats.mstats.mquantiles(sys_map, percentiles)
+
+        # Remove outliers in the systematic
+        clip = (sys_map > bin_edges[0]) & (sys_map < bin_edges[-1])
+        sys_map = sys_map[clip]
+        data_map = data_map[clip]
+
+        # Find the bin each pixel lands in
+        bins = np.digitize(sys_map, bin_edges) - 1
+
+        # Get the count in each bin in the
+        # systematic parameter
+        counts = np.bincount(bins)
+
+        # and the mean of the systematic value
+        # itself, and of the y and y^2 values
+        x = np.bincount(bins, weights=sys_map) / counts
+        y = np.bincount(bins, weights=data_map) / counts
+        y2 = np.bincount(bins, weights=data_map ** 2) / counts
+
+        # the var on the mean = var / count
+        yerr = np.sqrt((y2 - y**2) / counts)
```

Comment:
do you need to divide by counts?

Reply:
ohhh, I guess you're plotting the error on the mean here? in that case nvm

Reply:
yep, error on the mean.  which I think is the right thing to plot.